mlflow/tracking/_default_experiment/abstract_context.py

```diff
@@ -0,0 +1,37 @@
+from abc import ABCMeta, abstractmethod
```

Comment:
On filename, what's the reason to add "_" to "default_experiment"? This could be useful concept not just for Databricks. At least per @prakharg-msft 's comment it'd be useful for MSFT as well.